app/queries/find_products.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class FindProducts
+  def initialize(params)
+    @search_term = params[:search_term]
+    @category = params[:category]
+  end
+
+  def call
+    query = Product.all
+    query = apply_search(query)
+    query = by_category(query)
+
+    query.order(updated_at: :desc).includes(:category, :stocks, image_attachment: :blob)
+  end
+
+  private
+
+  def apply_search(query)
+    return query unless @search_term
```

Comment:
We should also check if @search_term isn't blank as we did in the `by_category` method 🤔 It works fine (because an empty string always matches) now but it adds this unnecessary condition to the SQL.